misp/misp.py

```diff
@@ -1,36 +1,73 @@
+from stix2 import Bundle, Identity, ThreatActor, IntrusionSet, Malware, Tool, Report, Indicator, Relationship, \
+    ExternalReference, TLP_WHITE, TLP_GREEN, \
     TLP_AMBER, TLP_RED
 
+CONNECTOR_IDENTIFIER = 'misp'
 
-class Misp:
-    def __init__(self, config, scheduler):
-        # Initialize config
-        self.config = config
-        self.scheduler = scheduler
 
-        # Initialize MISP
-        self.misp = PyMISP(self.config['misp']['url'], self.config['misp']['key'], False, 'json')
+class Misp:
+    def __init__(self):
+        # Get configuration
+        config_file_path = os.path.dirname(os.path.abspath(__file__)) + '/config.yml'
+        self.config = dict()
+        if os.path.isfile(config_file_path):
+            config = yaml.load(open(config_file_path), Loader=yaml.FullLoader)
+            self.rabbitmq_hostname = config['rabbitmq']['hostname']
+            self.rabbitmq_port = config['rabbitmq']['port']
+            self.rabbitmq_username = config['rabbitmq']['username']
+            self.rabbitmq_password = config['rabbitmq']['password']
+            self.config['name'] = config['misp']['name']
+            self.config['url'] = config['misp']['url']
+            self.config['key'] = config['misp']['key']
+            self.config['tag'] = config['misp']['tag']
+            self.config['untag_event'] = config['misp']['untag_event']
+            self.config['imported_tag'] = config['misp']['imported_tag']
+            self.config['interval'] = config['misp']['interval']
+            self.config['log_level'] = config['misp']['log_level']
+        else:
+            self.rabbitmq_hostname = os.getenv('RABBITMQ_HOSTNAME', 'localhost')
+            self.rabbitmq_port = os.getenv('RABBITMQ_PORT', 5672)
+            self.rabbitmq_username = os.getenv('RABBITMQ_USERNAME', 'guest')
+            self.rabbitmq_password = os.getenv('RABBITMQ_PASSWORD', 'guest')
+            self.config['name'] = os.getenv('MISP_NAME', 'MISP')
+            self.config['url'] = os.getenv('MISP_URL', 'http://localhost')
+            self.config['key'] = os.getenv('MISP_KEY', 'ChangeMe')
+            self.config['tag'] = os.getenv('MISP_TAG', 'OpenCTI: Import')
+            self.config['untag_event'] = os.getenv('MISP_UNTAG_EVENT', "True") == "True"
+            self.config['imported_tag'] = os.getenv('MISP_IMPORTED_TAG', 'OpenCTI: Imported')
+            self.config['interval'] = os.getenv('MISP_INTERVAL', 5)
+            self.config['log_level'] = os.getenv('MISP_LOG_LEVEL', 'info')
 
-        # Initialize OpenCTI client
-        self.opencti = OpenCTI(
-            self.config['opencti']['api_url'],
-            self.config['opencti']['api_key'],
-            os.path.dirname(os.path.abspath(__file__)) + '/misp.log',
-            True
+        # Initialize OpenCTI Connector
+        self.opencti_connector = OpenCTIConnectorHelper(
+            CONNECTOR_IDENTIFIER,
```

Comment:
Just use connector name no (config[connector][name])?

Reply:
There is a confusion with this configuration field "name". It is not the connector name but the name of the MISP instance (in case of using multiple MISP connectors). This name is used in the external reference : `source_name=self.config['name'],`.

Other connectors will not have any configuration of this kind. Furthermore, I think the connector identifier should be enforce by the developer directly in the source code and not a customizable config. So my choice would be to let the CONNECTOR_IDENTIFIER as this.